flyteplugins/go/tasks/plugins/k8s/kfoperators/mpi/mpi.go

```diff
@@ -110,36 +106,14 @@ func (mpiOperatorResourceHandler) BuildResource(ctx context.Context, taskCtx plu
 			return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "invalid TaskSpecification [%v], Err: [%v]", taskTemplate.GetCustom(), err.Error())
 		}
 
-		launcherReplicaSpec := kfMPITaskExtraArgs.GetLauncherReplicas()
-		if launcherReplicaSpec != nil {
-			// flyte commands will be passed as args to the container
-			err = common.OverrideContainerSpec(
-				launcherReplica.PodSpec,
-				kubeflowv1.MPIJobDefaultContainerName,
-				launcherReplicaSpec.GetImage(),
-				launcherReplicaSpec.GetResources(),
-				launcherReplicaSpec.GetCommand(),
-			)
-			if err != nil {
-				return nil, err
-			}
-			launcherReplica.RestartPolicy = common.ParseRestartPolicy(launcherReplicaSpec.GetRestartPolicy())
+		launcherReplicaSpec, err = common.ToReplicaSpecWithOverrides(ctx, taskCtx, kfMPITaskExtraArgs.GetLauncherReplicas(), kubeflowv1.MPIJobDefaultContainerName, true)
```

Comment:
Need to confirm if launcher pods should be treated as special and be non-interruptible.

Reply:
No clear evidence to suggest one way or the other. This pod is responsible for launching the jobs on the MPI workers, so will give it special treatment for now.